crates/core/component/auction/src/component/rpc.rs

```diff
@@ -22,4 +25,12 @@ impl Server {
 }
 
 #[tonic::async_trait]
-impl QueryService for Server {}
+impl QueryService for Server {
+    #[instrument(skip(self, request))]
+    async fn auction_state_by_id(
+        &self,
+        request: tonic::Request<AuctionStateByIdRequest>,
+    ) -> Result<tonic::Response<AuctionStateByIdResponse>, Status> {
+        todo!()
```

Comment:
Once we have modified the proto method signature to have `google.protobuf.Any`, we can: 
1. Pull a snapshot (read-only view of the chain state): `let storage = storage.latest_snapshot();`
2. Use `AuctionStoreRead::get_raw_auction` to get an `Option<Any>`
3. Inform the client if nothing was found, or
4. Prepare a tonic response if there was a match

I think one missing piece of context here, is that we want to have a minimal RPC that work for a variety of auction types and defer to clients to support (or not) serialization into a concrete type.

Reply:
I modified `get_raw_auction` to use `pbjson_types::Any` instead of `prost_types::Any` since the tonic response `AuctionStateByIdResponse` expects that, otherwise it throws mismatched type error. I think we can keep the streaming-based `auction_state_by_ids` unimplemented for now and circle back once we've sufficiently tested `auction_state_by_id`. 

I think having a minimal rpc where the client filters / serializes the raw data into a concrete type is a good design choice.